cli/azd/pkg/apphost/generate.go

```diff
@@ -439,7 +440,7 @@ func GenerateProjectArtifacts(
 	generatedFS := memfs.New()
 
 	projectFileContext := genProjectFileContext{
-		Name: projectName,
+		Name: label.LabelName(projectName),
```

Comment:
I think I'd prefer to do this at the layer where we pass in the name (see list of files below). My inclination here is that name sanitization/restriction enforcement is really important to be done at closest to the user interaction layer as possible.

- [environment_service_create.go](https://github.com/Azure/azure-dev/blob/4182b4138151b8fac4e9d5e6f6220b32dddace80/cli/azd/internal/vsrpc/environment_service_create.go#L73)
- [app_init.go](https://github.com/Azure/azure-dev/blob/4182b4138151b8fac4e9d5e6f6220b32dddace80/cli/azd/internal/repository/app_init.go#L176).

Reply:
moved the logic to `GetDefaultProjectName()`